### DISCUSSION TITLE
Getting started using Zim to learn python?

### DISCUSSION THREAD

**Discussion by tscahill** [zim-desktop-wiki/zim-desktop-wiki#2266] [Development]
There are several alternative phrasings around this topic question. I'd like to:
- start contributing to the development of Zim
- learn python
- Tweak and fiddle with the Zim and plugin code.

I've started messing with python and various IDEs. I have begun working through various online python resources. I think I might be a bit overwhelmed by all the different ways to set up the development environment. I've moved through a dozen or so Linux distributions over the last 12 months  So at this point, my real ask is:

How do you have your Zim programming environment configured for working with the Zim main code and the various plugins. Is there a preferred Integrated Development Environment?

**Comment by Toggoren** [comment]
Just use ``PyCharm``. 
Community version is free.

**Reply by tscahill** [reply]
Thanks Toggoren. I thought PyCharm was only available in paid versions, but I've downloaded the opensource program and so far so good.
